heroic-core/src/main/java/com/spotify/heroic/servlet/MandatoryClientIdFilter.java

```diff
@@ -21,32 +21,36 @@
 
 package com.spotify.heroic.servlet;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.common.base.Strings;
+import com.spotify.heroic.ws.ErrorMessage;
+import com.spotify.heroic.ws.MandatoryClientIdErrorMessage;
 import java.io.IOException;
-import javax.servlet.Filter;
 import javax.servlet.FilterChain;
-import javax.servlet.FilterConfig;
 import javax.servlet.ServletException;
 import javax.servlet.ServletRequest;
-import javax.servlet.ServletResponse;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.ws.rs.core.Response.Status;
 
 /**
  * Rejects anonymous requests. That is, requests to any API endpoint that are
- * missing a non-null X-Client-Id HTTP header.
+ * missing a non-null X-Client-Id HTTP header.<p></p>
+ * *Note* that this @SuppressWarnings has to go here even though it's just for the doFilter
+ * method, else the JavaDoc for it doesn't render. Weird.
  */
-// Note that this @Suppress has to go here even though it's just for the doFilter
-// method, else the JavaDoc for it doesn't render. Weird.
 @SuppressWarnings("checkstyle:LineLength")
-public class MandatoryClientIdFilter implements Filter {
+public class MandatoryClientIdFilter extends SimpleFilter {
 
     public static final String X_CLIENT_ID_HEADER_NAME = "X-Client-Id";
+    public static final String ERROR_MESSAGE_TEXT =
+            "This anonymous request has been rejected. Please add a 'x-client-id' " +
+                    "HTTP header to your request. Help can be found in the" +
+                    " #monitoring-users Slack channel.";
```

Comment:
This is on the public repo. We probably don't want Spotify specific messages.

Reply:
good point.